api/src/test/java/uk/gov/dhsc/htbhf/claimant/reporting/ClaimantCategoryCalculatorTest.java

```diff
@@ -0,0 +1,108 @@
+        Claimant claimant = aValidClaimantBuilder()
+                .expectedDeliveryDate(expectedDeliveryDate)
+                .dateOfBirth(LocalDate.now().minusYears(20))
+                .build();
+        given(pregnancyEntitlementCalculator.isEntitledToVoucher(any(), any())).willReturn(true);
+
+        String claimantCategory = claimantCategoryCalculator.determineClaimantCategory(claimant, datesOfBirthOfChildren, LocalDateTime.now());
+
+        LocalDate now = LocalDate.now();
+        assertThat(claimantCategory).isEqualTo("Pregnant with passported children");
+        ArgumentCaptor<LocalDate> argumentCaptor = ArgumentCaptor.forClass(LocalDate.class);
+        verify(pregnancyEntitlementCalculator).isEntitledToVoucher(eq(expectedDeliveryDate), argumentCaptor.capture());
+        assertThat(argumentCaptor.getValue()).isAfterOrEqualTo(now);
+    }
+
+    @Test
+    void shouldMatchClaimantToPregnantWithNoChildren() {
+        LocalDate expectedDeliveryDate = LocalDate.now().plusMonths(2);
+        Claimant claimant = aValidClaimantBuilder()
+                .expectedDeliveryDate(expectedDeliveryDate)
+                .dateOfBirth(LocalDate.now().minusYears(20))
+                .build();
+        given(pregnancyEntitlementCalculator.isEntitledToVoucher(any(), any())).willReturn(true);
+
+        String claimantCategory = claimantCategoryCalculator.determineClaimantCategory(claimant, emptyList(), LocalDateTime.now());
+
+        LocalDate now = LocalDate.now();
+        assertThat(claimantCategory).isEqualTo("Pregnant and no passported children");
+        ArgumentCaptor<LocalDate> argumentCaptor = ArgumentCaptor.forClass(LocalDate.class);
+        verify(pregnancyEntitlementCalculator, times(2)).isEntitledToVoucher(eq(expectedDeliveryDate), argumentCaptor.capture());
+        assertThat(argumentCaptor.getValue()).isAfterOrEqualTo(now);
+    }
+
+    @Test
+    void shouldMatchClaimantToNotPregnantWithChildren() {
+        LocalDate expectedDeliveryDate = null;
+        List<LocalDate> datesOfBirthOfChildren = singletonList(LocalDate.now().minusYears(2));
+        Claimant claimant = aValidClaimantBuilder()
+                .expectedDeliveryDate(expectedDeliveryDate)
+                .dateOfBirth(LocalDate.now().minusYears(20))
+                .build();
+        given(pregnancyEntitlementCalculator.isEntitledToVoucher(any(), any())).willReturn(false);
+
+        String claimantCategory = claimantCategoryCalculator.determineClaimantCategory(claimant, datesOfBirthOfChildren, LocalDateTime.now());
+
+        LocalDate now = LocalDate.now();
+        assertThat(claimantCategory).isEqualTo("Not pregnant with passported children");
+        ArgumentCaptor<LocalDate> argumentCaptor = ArgumentCaptor.forClass(LocalDate.class);
+        verify(pregnancyEntitlementCalculator, times(3)).isEntitledToVoucher(eq(expectedDeliveryDate), argumentCaptor.capture());
+        assertThat(argumentCaptor.getValue()).isAfterOrEqualTo(now);
```

Comment:
You should refactor out all these duplicate calls to check the date (passing in the number of times of course).

Reply:
Code has been refactored so there are no longer multiple calls.